types/vote.go

```diff
@@ -276,8 +276,8 @@ func (vote *Vote) ValidateBasic() error {
 		return errors.New("invalid Type")
 	}
 
-	if vote.Height < 0 {
-		return errors.New("negative Height")
+	if vote.Height <= 0 {
```

Comment:
What's the impact of votes having a height of zero in previous versions, like v0.37 and v0.34?

Reply:
A vote with height of 0 would mean there is a bug.
I just checked `Vote::ValidateBasic` in `v0.37.x` and in `v0.34.x` and we can see the same problem with this "if". It is likely that we never have a vote with height 0 in production in `v0.37.x` or in `v0.34.x`, and so this check wouldn't be catching any existing problem? (wishful thinking?).
Anyway, the fact that UTs pass in `v0.37.x` and in `v0.34.x` shows there is a gap in UTs.

Here's what I propose we can do: in order not to delay this PR, let us:
* Revert this change in this PR
* Open an issue to fix it on all maintained branches, and add a UT that fails without this change and passes with this change (TDD-style)
What do you think?

Reply:
If the correct behaviour is to reject votes with zero height, we should keep it. The same behaviour should be applied to older versions, but it doesn't seem urgent right now to do so, especially since we make no compatibility guarantees across "major" releases.

Reply:
Thanks, makes sense. I added an extra point in #9916 to deal with this in `v0.34.x` and `v0.37.x`. We can convert it into an issue at any time.